examples/config.yaml

```diff
@@ -1,116 +1,215 @@
       namespace: argocd
+    # destination of the application
     destination:
+      # on which server to deploy
       server: https://kubernetes.default.svc
+      # in which namespace to deploy
       namespace: default
+    # source of the application
     source:
+      # which values file to use in the Helm chart
       valuefile: values.yaml
-      path: helm/golang-demo
-      repoURL: https://github.com/ironcore864/golang-demo.git
-- name: kube-prometheus-dev
+      # path of the Helm chart
+      path: charts/DO_CHANGE_ME
+      # Helm chart repo URL
+      repoURL: https://github.com/DO_CHANGE_ME/DO_CHANGE_ME.git
+# name of the instance with kube-prometheus
+- name: dev
   plugin:
+    # kind of the plugin
     kind: kube-prometheus
+    # version of the plugin
     version: 0.2.0
+  # options for the plugin
   options:
-    create_namespace: true
+    # need to create the namespace or not, default: false
+    create_namespace: false
+    # Helm repo information
     repo:
+      # name of the Helm repo
       name: prometheus-community
+      # url of the Helm repo
       url: https://prometheus-community.github.io/helm-charts
+    # Helm chart information
     chart:
+      # name of the chart
       chart_name: prometheus-community/kube-prometheus-stack
+      # release name of the chart
       release_name: dev
+      # k8s namespace where kube-prometheus will be installed
       namespace: monitoring
+      # whether to wait for the release to be deployed or not
       wait: true
+      # the time to wait for any individual Kubernetes operation (like Jobs for hooks). This defaults to 5m0s
       timeout: 5m
+      # whether to perform a CRD upgrade during installation
       upgradeCRDs: true
 - name: devlake
   plugin:
+    # kind of the plugin
     kind: devlake
+    # version of the plugin
     version: 0.2.0
+
+# name of the instance with jenkins
 - name: jenkins-dev # "you need to create the ~/data/jenkins-volumes/ directory first and chown this directory to 1000:1000"
   plugin:
+    # kind of the plugin
     kind: jenkins
+    # version of the plugin
     version: 0.2.0
+  # options for the plugin
   options:
-    create_namespace: true
+    # need to create the namespace or not, default: false
+    create_namespace: false
+    # Helm repo information
     repo:
+      # name of the Helm repo
       name: jenkins
+      # url of the Helm repo
       url: https://charts.jenkins.io
+    # Helm chart information
     chart:
+      # name of the chart
       chart_name: jenkins/jenkins
+      # release name of the chart
       release_name: dev
+      # k8s namespace where jenkins will be installed
       namespace: jenkins
+      # whether to wait for the release to be deployed or not
       wait: true
+      # the time to wait for any individual Kubernetes operation (like Jobs for hooks). This defaults to 5m0s
       timeout: 5m
+      # whether to perform a CRD upgrade during installation
```

Comment:
As the OP suggested, this `examples/config.yaml` file has grown too big for new users to get a "quick start". 

As a workaround, maybe we can create multiple "examples", like:

- `examples/golang-webapp-ci-cd.yaml`: github repo scaffolding + github actions + argocd + argocd app
- `examples/golang-webapp-trello.yaml`: github repo scaffolding + github actions + trello
- `examples/observability.yaml`: devlake + prometheus
- `examples/gitlab-ci.yaml`: gitlab ci (TODO: here maybe we can add gitlab repo scaffolding later as well)
- `examples/jenkins.yaml`: jenkins

Credit to @daniel-hutao who suggested that we should have some "scenarios" for users to begin quickly.